examples/persistent_oauth.py

```diff
@@ -0,0 +1,69 @@
+"""Bring Your Own Identity Provider with fined grained OAuth scopes is currently public preview on
+Databricks in AWS. databricks-sql-connector supports user to machine OAuth login which means the
+end user has to be present to login in a browser which will be popped up by the Python process. You
+must enable OAuth in your Databricks account to run this example. More information on how to enable
+OAuth in your Databricks Account in AWS can be found here:
+
+https://docs.databricks.com/administration-guide/account-settings-e2/single-sign-on.html
+
+Pre-requisites:
+- You have a Databricks account in AWS.
+- You have configured OAuth in Databricks account in AWS using the link above.
+- You have installed a browser (Chrome, Firefox, Safari, Internet Explorer, etc) that will be
+  accessible on the machine for performing OAuth login.
+
+For security, databricks-sql-connector does not persist OAuth tokens automatically. Hence, after
+the Python process terminates the end user will have to log-in again. We provide APIs to be
+implemented by the end user for persisting the OAuth token. The SampleOAuthPersistence reference
+shows which methods you may implement.
+
+For this example, the DevOnlyFilePersistence class is provided. Do not use this in production.
+
+Bring Your Own Identity Provider is in public preview. You can monitor these two links to find out
+when it will become generally available:
+
+  1. https://docs.databricks.com/administration-guide/account-settings-e2/single-sign-on.html 
+  2. https://docs.databricks.com/dev-tools/python-sql-connector.html
+"""
+
+import os
+from typing import Optional
+
+from databricks import sql
+from databricks.sql.experimental.oauth_persistence import OAuthPersistence, OAuthToken, DevOnlyFilePersistence
```

Comment:
could you please also mention that the API (interface) for OAuthPersistence are experimental and evolve over time. I think the package name `databricks.sql.experimental.oauth_persistence` indicates that but it would be great if we could also explicitly say so.

Reply:
Updated in [f8a81e8](https://github.com/databricks/databricks-sql-python/pull/52/commits/f8a81e8605448c48f65e769da79f8086d4d3d82b)

Thanks!